include/roboteam_utils/Rectangle.h

```diff
@@ -89,27 +89,12 @@ class Rectangle {
      * @return vector with all of the points where the line segment intersects the rectangle
      */
     [[nodiscard]] std::vector<Vector2> intersects(const LineSegment &line) const;
-    /**
-     * @brief Checks whether a given line intersects with `this`.
-     * Not the most efficient and may return the same point twice if the line 'exactly' intersects with the corner.
-     * In general prefer using LineSegments over Lines if you need intersections often.
-     * @param line line segment to check for
-     * @return vector with all of the points where the line segment intersects the rectangle
-     */
-
-    [[nodiscard]] std::vector<Vector2> intersects(const Line &line) const;
```

Comment:
Why is this removed?

Reply:
For this you need to compute the intersection between a Line and LineSegment. And since the Line and LineSegment class are totally redesigned, I removed the Line and LineSegment intersect methods and did not add it again, since I do not spend time adjusting dead code.